src/hierarchyFinder.js

```diff
@@ -0,0 +1,62 @@
+var _ = require('lodash');
+
+module.exports = {};
+
+var hasName = function(r) {
+  return r.name;
+};
+
+var isDefined = function(r) {
```

Comment:
What about removing this function and using _.isObject? I think all the records coming through should be objects, and that way we can use a built in Lodash function.